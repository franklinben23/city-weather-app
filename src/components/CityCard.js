import React from 'react';

const CityCard = (props) => {
  const { cityImage, cityCountry, cityName } = props;
  return (
    <div className="city-card">
      <div className="image-cont">
        <img src={cityImage} alt="city-weather" className="city-img" />
      </div>
      <p className="city-span">
        {' '}
        {cityName}
        :
        {' '}
        {cityCountry}
      </p>
    </div>
  );
};

export default CityCard;
