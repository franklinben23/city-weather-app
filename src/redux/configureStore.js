import { combineReducers, createStore, applyMiddleware } from 'redux';
import Thunk from 'redux-thunk';
import logger from 'redux-logger';
import CityWeather from './cities/cities';

const rootReducer = combineReducers({
  cityWeathers: CityWeather,
});

const store = createStore(rootReducer, applyMiddleware(Thunk, logger));

export default store;
