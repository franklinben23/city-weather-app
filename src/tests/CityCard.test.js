import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import CityCard from '../components/CityCard';
import '@testing-library/jest-dom';
import configureStore from '../redux/configureStore';

describe('Testing Mission', () => {
  it('Testing Mission component snapshot', () => {
    const element = {
      id: 'key1',
      cityName: 'TEST ROCKET',
      cityCountry: 'TEST DESCRIPTION',
      cityImage: 'https://imgur.com/DaCfMsj.jpg',
    };
    const testRenderer = TestRenderer.create(
      <Provider store={configureStore}>
        <CityCard
          key="key1"
          cityName={element.cityName}
          cityCountry={element.cityCountry}
          cityImage={element.cityImage}
        />
      </Provider>,
    ).toJSON();
    expect(testRenderer).toMatchSnapshot();
  });
});
