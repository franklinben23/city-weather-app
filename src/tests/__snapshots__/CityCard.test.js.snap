// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Mission Testing Mission component snapshot 1`] = `
<div
  className="city-card"
>
  <div
    className="image-cont"
  >
    <img
      alt="city-weather"
      className="city-img"
      src="https://imgur.com/DaCfMsj.jpg"
    />
  </div>
  <p
    className="city-span"
  >
     
    TEST ROCKET
    :
     
    TEST DESCRIPTION
  </p>
</div>
`;
